tracks/x86-64-assembly/exercises/two-fer/mentoring.md

```diff
@@ -0,0 +1,94 @@
+### Reasonable solutions
+
+Using a hand-rolled loop:
+
+```nasm
+    movzx edx, byte [rdi]
+    test dl, dl
+    je .loop_end
+.loop_start:
+    add rdi, 1
+    mov byte [rsi], dl
+    add rsi, 1
+    movzx edx, byte [rdi]
+    test dl, dl
+    jne .loop_start
+.loop_end:
+```
+
+Using the `rep movsb` instructions:
+
+```nasm
+    mov rdi, rsi
+    lea rsi, [rel symbol]
+    mov rcx, symbol_len
+    rep movsb
+```
+
+### Common errors
+
+- Forgetting to save registers. "Registers `%rbp`, `%rbx` and `%r12` through `%r15` 'belong' to the calling function and the called function is required to preserve their values. In other words, a called function must preserve these registers’ values for its caller" (Matz, Hubička, Jaeger, & Mitchell, 2014, p. 15). Depending on which registers the caller is using, the tests might still pass.
```

Comment:
```suggestion
- Forgetting to save registers. 

  > Registers `%rbp`, `%rbx` and `%r12` through `%r15` 'belong' to the calling function and the called function is required to preserve their values. In other words, a called function must preserve these registers’ values for its caller
  > — Matz, Hubička, Jaeger, & Mitchell, 2014, p. 15
  
  Depending on which registers the caller is using, the tests might still pass.
```

I think this was the intention? This is what it will look like:

### Common errors

- Forgetting to save registers. 

  > Registers `%rbp`, `%rbx` and `%r12` through `%r15` 'belong' to the calling function and the called function is required to preserve their values. In other words, a called function must preserve these registers’ values for its caller
  > — Matz, Hubička, Jaeger, & Mitchell, 2014, p. 15
  
  Depending on which registers the caller is using, the tests might still pass.